workflow/controller/controller.go

```diff
@@ -705,7 +705,13 @@ func (wfc *WorkflowController) processNextItem(ctx context.Context) bool {
 		}
 		if doPodGC {
 			for podName := range woc.completedPods {
-				woc.controller.queuePodForCleanup(woc.wf.Namespace, podName, deletePod)
+				if woc.execWf.Spec.PodGC.DeleteDelaySeconds != nil {
```

Comment:
This logic is duplicated between here and the operator, I am usually in the "refactor after 3 times" camp but can be swayed to do something now.